src/Week.jsx

```diff
@@ -106,6 +107,16 @@ export default class Week extends PureComponent {
     });
   }
 
+  // generate the props required for Day to block specific hours.
+  generateHourLimits() {
+    const { availableHourRange } = this.props;
+    return { top: availableHourRange.start * HOUR_IN_PIXELS, // top blocker
+      bottom: availableHourRange.end * HOUR_IN_PIXELS,
+      bottomHeight: (24 - availableHourRange.end) * HOUR_IN_PIXELS, // bottom height
+      difference: (availableHourRange.end - availableHourRange.start) * HOUR_IN_PIXELS,
+    };
+  }
```

Comment:
In an effort to keep the props for `<Day>` cleaner, would it make sense to do this inside Day.jsx? Don't worry about the additional computation needed, I think it's neglible in the grand scheme of things.

Reply:
I did calculate it in `<Day>` initially but thought there was quite a bit of duplicated code (e.g. bottom distance was used a few times and I didn't like the idea of writing a function just for a simple calculation) so I thought to pass it down as props. Furthermore, it's the same constraints for all `<Day>` objects anyway. 

I can still move it back to `<Day>` if we want to prioritise a cleaner props however.

Reply:
Nah, this is fine. Thanks for the explanation.